src/zustand/tokens.ts

```diff
@@ -0,0 +1,43 @@
+import type { Address } from 'viem'
+import { useSyncExternalStoreWithTracked } from '~/hooks/useSyncExternalStoreWithTracked'
+import { createStore } from './utils'
+
+export type TokensState = {
+  tokens: Address[]
+}
```

Comment:
We will need to make sure this works across accounts too. 
```suggestion
export type TokensState = {
  tokens: {
    [account: Address]: Address[]
  }
}
```